File: v0dot2/project_environment_all/js_unittest/test_new_round.js
const { DateTime } = require('luxon');
const GenericWorld = require('../path/to/GenericWorld');

describe('TestGenericWorld', () => {
  let genericWorld;

  beforeEach(() => {
    genericWorld = new GenericWorld();
  });

  test('test_new_round', () => {
    // Mock necessary dependencies
    genericWorld.logger = { warning: jest.fn() };
    genericWorld.build_arena = jest.fn().mockReturnValue([Array(5).fill(0), [], []]);
    genericWorld.active_agents = Array(2).fill().map(() => ({ start_round: jest.fn() }));

    // Call the method
    genericWorld.new_round();

    // Assertions on method behavior
    expect(genericWorld.running).toBe(false);
    expect(genericWorld.logger.warning).toHaveBeenCalledWith('New round requested while still running');
    expect(genericWorld.end_round).toHaveBeenCalledTimes(1);
    expect(genericWorld.round).toBe(1);
    expect(genericWorld.step).toBe(0);
    expect(genericWorld.bombs).toEqual([]);
    expect(genericWorld.explosions).toEqual([]);
    expect(genericWorld.round_id).toBe(`Round 01 (${DateTime.now().toFormat('yyyy-MM-dd HH-mm-ss')})`);
    expect(genericWorld.build_arena).toHaveBeenCalledTimes(1);
    genericWorld.active_agents.forEach(agent => {
      expect(agent.start_round).toHaveBeenCalledTimes(1);
    });
    expect(genericWorld.replay.round).toBe(1);
    expect(genericWorld.replay.arena).toEqual(Array(5).fill().map(() => Array(5).fill(0)));
    expect(genericWorld.replay.coins).toEqual([]);
    expect(genericWorld.replay.agents).toEqual([]);
    expect(genericWorld.replay.actions).toEqual({});
    expect(genericWorld.replay.permutations).toEqual([]);
  });
});

File: previous_versions/v0dot3/project_environment_all/py_unittest/test_new_round.py
from datetime import datetime
import numpy as np
import unittest
from unittest.mock import Mock
from environment_all import GenericWorld

class TestGenericWorld(unittest.TestCase):
    def setUp(self):
        self.generic_world = GenericWorld()
    
    def test_new_round(self):
        # Mock necessary dependencies
        self.generic_world.logger = Mock()
        self.generic_world.build_arena = Mock(return_value=(np.zeros((5, 5)), [], []))
        self.generic_world.active_agents = [Mock() for _ in range(2)]
        for agent in self.generic_world.active_agents:
            agent.start_round = Mock()
        
        # Call the method
        self.generic_world.new_round()
        
        # Assertions on method behavior
        self.assertFalse(self.generic_world.running)
        self.generic_world.logger.warning.assert_called_with('New round requested while still running')
        self.generic_world.end_round.assert_called_once()
        self.assertEqual(self.generic_world.round, 1)
        self.assertEqual(self.generic_world.step, 0)
        self.assertEqual(self.generic_world.bombs, [])
        self.assertEqual(self.generic_world.explosions, [])
        self.assertEqual(self.generic_world.round_id, f"Round 01 ({datetime.now().strftime('%Y-%m-%d %H-%M-%S')})")
        self.generic_world.build_arena.assert_called_once()
        for agent in self.generic_world.active_agents:
            agent.start_round.assert_called_once()
        self.assertEqual(self.generic_world.replay['round'], 1)
        self.assertTrue(np.array_equal(self.generic_world.replay['arena'], np.zeros((5, 5))))
        self.assertEqual(self.generic_world.replay['coins'], [])
        self.assertEqual(self.generic_world.replay['agents'], [])
        self.assertEqual(self.generic_world.replay['actions'], dict())
        self.assertEqual(self.generic_world.replay['permutations'], [])